objects/oWeakling/Draw_0.gml

```diff
@@ -19,22 +25,53 @@ if instance_exists(_weakling_inst) {
 	}
 }
 
+// Sync oWeakling's sprite animation frame with the progress bar. This allows
+// the animation to speed up as we purchase speed upgrades. 
+// TODO: How can we write this to account for possible range-of-motion upgrades?
 // progress bar instance ID
-_inst = inst_61B8692F
-
-// TODO: Generalize this so that it works for more than one frame, so we
-// can offer an upgrade to range-of-motion. 
+var _inst = inst_61B8692F
 if instance_exists(_inst) {
+
+	var _range_of_motion = 4
```

Comment:
TODO: Turn this into an upgrade, otherwise we won't see benefits.

Reply:
Fixed in https://github.com/ggrammar/IdleLearning3/pull/6/commits/213d663e59ff905a005a06532febf847d19a1354